### PR TITLE
Search for go.mod in the provider folder as well

### PR DESCRIPTION
**Pull request by praneetloke**
I think this is needed for the new module structure in the provider repos to work.

### CODE REVIEW

File: pkg/tfgen/gomod.go [hunk 1]
```diff
@@ -33,7 +33,12 @@ func LoadGoMod() (*modfile.File, error) {
 
 	moduleRoot := findModuleRoot(exePath)
 	if moduleRoot == "" {
-		return nil, errors.New("cannot find module root")
+		// Some provider repos have a "provider" module, rather than a
+		// module at the root of the repo.
+		moduleRoot = findModuleRoot(exePath + "provider")
```
Comment (praneetloke) [inline pkg/tfgen/gomod.go]:
```suggestion
		moduleRoot = findModuleRoot(exePath + "/provider")
```

File: pkg/tfgen/gomod.go [hunk 2]
```diff
@@ -33,7 +33,12 @@ func LoadGoMod() (*modfile.File, error) {
 
 	moduleRoot := findModuleRoot(exePath)
 	if moduleRoot == "" {
-		return nil, errors.New("cannot find module root")
+		// Some provider repos have a "provider" module, rather than a
+		// module at the root of the repo.
+		moduleRoot = findModuleRoot(exePath + "/provider")
```
Comment (lukehoban) [inline pkg/tfgen/gomod.go]: Should probably use filepath.Join.

**Comment by praneetloke** [comment]
@lukehoban unless I am missing something, I think this is safe to land regardless, as this PR only falls-back to searching in the `/provider` repo, if a `go.mod` couldn't be found at the root, which is the old/current way.

**Comment by EvanBoyle** [comment]
@lukehoban @praneetloke ideally these changes would all be made atomically, but that isn't really possible. Every provider needs to be updated to consume the new modules structure from pulum/pulumi, and the providers need to be broken down into multi-module repos which inclue top level go.mod files for /provider and /sdk.

**Pull request merged**